xdem/terrain.py

```diff
@@ -67,6 +67,9 @@ def _get_quadric_coefficients(
                 elif edge_method_n == 1:
                     row_indexer = (row + k) % dem.shape[0]
                     col_indexer = (col + j) % dem.shape[1]
+                else:
+                    row_indexer = row + k
```

Comment:
Are you sure this won't overflow at any point?

Reply:
Not sure I got this. You mean you'd prefer `elif edge_method_n == 2`?

Reply:
It may be just me not remembering my own code haha, but what happens if `row == dem.shape[0]` and `k` is nonzero? Then that would result in an `IndexError`

Reply:
It is the non-remembering haha, you wrote a loop further up to avoid that issue :wink::
```python
  if edge_method_n == 2:
            if (row < 1) or (row > (dem.shape[0] - 2)) or (col < 1) or (col > (dem.shape[1] - 2)):
                continue
```
And tests would have picked that up for sure. 
So all ready to be merged :+1:

Reply:
Past me is great and so are you!